### DISCUSSION TITLE
Lay-Z-Spa Vancouver - pump S100202 - 4 wire *and* 6 wire!

### DISCUSSION THREAD

**Discussion by ghost** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#374] [General]
A few months ago I bought a bargain Lay-Z-Spa Vancouver, with the post-2020 square airjet pump labelled S100202.

Having found this repo, and seen loads of people with square pumps having 6 wires to the DSP and 6 to the CIO, I soldered up a D1 Mini R2, built the firmware, and went to fit it. Imagine my surprise when I found that the connector from the CIO has 6 wires, but the DSP connector only has 4!

I've successfully managed to get the display powered up (pins 1 & 2 on the DSP connector are +5V and GND), but beyond having all the LEDS lit up I can't get anything else to happen.have tried a variety of GPIO assignments and swapping pins, to no avail.

If anyone has the tiniest idea, I'm all ears!

![IMG_20220910_210502](https://user-images.githubusercontent.com/1202469/189500303-851eb69b-1d3e-409c-a72a-49a550993cef.jpg)
![IMG_20220910_210425](https://user-images.githubusercontent.com/1202469/189500313-e9f15687-0b1b-4634-8770-b4a095e7e50f.jpg)

**Comment by wyliecoyoteuk** [comment]
Maybe you need to treat it as a 4 wire system. Obviously the other 2 wires are not being used.

**Reply by visualapproach** [reply]
Yes I remember someone else having this strange setup. Maybe you can find the post in this forum. Agree with @wyliecoyoteuk

**Reply by wyliecoyoteuk** [reply]
I think this it. comments later in other threads suggest he may have got it working
https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/138#discussioncomment-1724558

**Reply by ghost** [reply]
Aha! Thanks for the pointer. I'll look into that.

Notably when the unit was switched on, there was a hum (speaker?) that gently changed if I figured anything within the web UI, so something was definitely getting back down to the CIO.

**Comment by ghost** [comment]
Cracked open the display panel and found an ESP-WROOM-02UC embedded in waterproof gel inside.

Pinout for the connector is labelled (see photo):

Pin 1 (black) +5V
Pin 2 (yellow) GND
Pin 3 (red) WT so presumably TX from this end
Pin 4 (white) WR so again presumably RX

There's an unconnected header marked LA. The display board dates from 2019, so I think you're right on the "mash stuff we've got together" comment in the linked thread.

![IMG_20220911_124144__01](https://user-images.githubusercontent.com/1202469/189526211-94e67b26-f2b8-41f6-9a78-4971304fda1d.jpg)

![IMG_20220911_124127](https://user-images.githubusercontent.com/1202469/189526207-5fc10419-951b-4d4f-8d7b-7b34a29b6737.jpg)

**Comment by ghost** [comment]
Having delved into the innards of the pump and uncovered the control module, it has the following text on the board:

Q21_AIRSPA_2Q_BOOMOO
2019.09.23

Web searches turn up no matches for that - those who are familiar with such things may have an idea what that indicates... maybe!

**Reply by visualapproach** [reply]
It would be interesting to sniff the serial com between the esp wroom module and the cio. It's definitely an extra possibility to hack that.  Not that I have the time but still.
Tell us how it goes when you try the 4-wire version on the ordinary cable.

**Comment by ghost** [comment]
Hello chaps

It's been a while, and with one thing and another I didn't get chance until just now to have a play with this. Connecting a USB logic analyser, capturing the data from all lines, feeding into PulseView pulls out a decent amount of info.

Wire 1 is 5V and wire 2 is GND, which we knew. Wires 3 & 4 are data lines.
I'm trying now to work out what the protocol is - either UART or I2C. The fact that the ESP board has WR and WT printed on it suggests UART, if it was I2C I'd expect something like CL & DA or similar. I've yet to work out the magic baud rate/data/parity/stop bit combination that doesn't throw frame errors at the end of each frame though.

**Reply by ghost** [reply]
It's serial (like the others). In the next day or two I'll try to operate every single button to see what I can capture.

**Reply by visualapproach** [reply]
Cool!

**Reply by ghost** [reply]
OK... so I now have a logic analyser dump at 24MHz, 8 bits. I've very definitely hit the limit of what I can understand from it now!

I did a sequence of:

Unit power on
Wait for display to finish self-test
Press & hold power button
Press and hold lock to lock
Press and hold lock to unlock
Temp up, temp up, temp up, wait for display to settle
Temp down, temp down, temp down, wait for display to settle
Bubbles on (max)
Bubbles off
Bubbles on (reduced)
Bubbles off
Power off
Couldn't trial pump on or heat on as it's disconnected from the tub and sitting by my desk!

Would it help if I was to upload the dump? If so, in which format? Thanks :)

**Reply by visualapproach** [reply]
Yeah can you zip the .csv file and post here? Worst case just dump the exported text as hex here. If using selae you can post the files here if it's possible.

**Reply by ghost** [reply]
It borked exporting a CSV (it reached > 2GB in size and stopped writing!), so here's a zip with the .sr file from Pulseview. It is indeed using a salae clone analyser.

Lines 1 thru 4 are PWR (5V), GND, data 1 and data 2 respectively.
Lines 5 and 6 are connected to the CIO lines 5 & 6, just because. There's a 4kHz signal on line 5 for half a second or so a few seconds after power on, and a couple more signals which I think correlate with button pushes.

I've left the UART and I2C decoders on the trace, so see what you make of them!

[hot-tub-mon.sr.zip](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/files/10668109/hot-tub-mon.sr.zip)

**Reply by visualapproach** [reply]
I cannot read .sr files. Can you export to a generic format? Or .logicdata

**Reply by ghost** [reply]
Yep, here you go, generic binary export, 8ch/24MHz direct off the reader.
[hot-tub-mon-8ch-24MHz.bin.zip](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/files/10731372/hot-tub-mon-8ch-24MHz.bin.zip)

**Reply by visualapproach** [reply]
Sorry, but I don't know what to do with 2 gigs of unknown data. By generic, I meant decoded SPI to a .CSV file with timestamps, or excel or something. Like this:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Thomas/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Thomas/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



Time [s] | Packet ID | MOSI | MISO
-- | -- | -- | --
1.305080000000000 | 0 | '255' (0xFF) | '128'   (0x80)
1.305545000000000 | 1 | '255' (0xFF) | '2' (0x02)
1.306015000000000 | 2 | '255' (0xFF) | '3'   (0x03)
1.306400000000000 | 2 | '255' (0xFF) | '255' (0xFF)
1.306780000000000 | 2 | '255' (0xFF) | '255'   (0xFF)
1.307165000000000 | 2 | '255' (0xFF) | '255' (0xFF)
1.307545000000000 | 2 | '255' (0xFF) | '255'   (0xFF)
1.307930000000000 | 2 | '255' (0xFF) | '255' (0xFF)
1.308310000000000 | 2 | '255' (0xFF) | '255'   (0xFF)
1.308695000000000 | 2 | '255' (0xFF) | '255' (0xFF)
1.309075000000000 | 2 | '255' (0xFF) | '255'   (0xFF)
1.309460000000000 | 2 | '255' (0xFF) | '255' (0xFF)
1.309840000000000 | 2 | '255' (0xFF) | '255'   (0xFF)
1.310310000000000 | 3 | '255' (0xFF) | '241' (0xF1)
1.327070000000000 | 4 | '255' (0xFF) | B   (0x42)
1.327450000000000 | 4 | '255' (0xFF) | '232' (0xE8)
1.327860000000000 | 4 | '255' (0xFF) | L   (0x4C)
1.341825000000000 | 5 | '255' (0xFF) | B (0x42)
1.342205000000000 | 5 | '255' (0xFF) | '204'   (0xCC)
1.342605000000000 | 5 | '255' (0xFF) | H (0x48)



</body>

</html>

**Comment by collse** [comment]
@greem did you ever get this one to work?

**Reply by ghost** [reply]
Nope. It hasn't been powered on for well over a year now, sorry.

**Reply by collse** [reply]
I ordered a different control panel and hopefully that will work - wifi would be obsolete anyway in that case

**Reply by collse** [reply]
ok, so I ordered another controller module without wifi (https://www.thespaworks.co.uk/product-page/n26) - which is a 6 pin as well. replaced the existing module - only 4 screws on the inside, then extended the wiring on the actual pcb to accommodate the module sitting in the top of the housing where no splash water should reach.

set hardware config to: MIAMI2021
CIO -  6 wire, 2021 (air)
DSP -  6 wire, 2021 (air)
PCBv2B

Happy days, everything is working as expected - so it is definitely the issue with the onboard wifi config on the S100202 controller as the S100102 works without issue with the remaining parts of the pump (as I previously expected they just replace the display depending on wifi/non-wifi model).
